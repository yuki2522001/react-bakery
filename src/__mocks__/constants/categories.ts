import { CategoryProps } from "@common-types/category";

const CATEGORY_MOCKING_LIST: CategoryProps[] = [
  {
    id: "1",
    name: "Cake"
  },
  {
    id: "2",
    name: "Breads"
  }
]

export { CATEGORY_MOCKING_LIST };
