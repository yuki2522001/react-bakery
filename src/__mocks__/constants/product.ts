import { Product } from "@common-types/product";
import url from "../../assets/images/products/black-forest-cake.jpg";

const PRODUCT_MOCKING: Product = {
  id: "1",
  categoryId: "1651924356334",
  name: "Cheese cake",
  price: 20,
  quantity: 5,
  description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
  images: [
    url
  ]
}

const PRODUCT_MOCKING_LIST: Product[] = [
  {
    id: "1",
    categoryId: "1651924356334",
    name: "Cheese cake",
    price: 20,
    quantity: 5,
    description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
    images: [
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg"
    ]
  },
  {
    id: "2",
    categoryId: "1651924356334",
    name: "Amaretti Cookie",
    price: 10,
    quantity: 5,
    description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
    images: [
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg"
    ]
  }
]

export { PRODUCT_MOCKING, PRODUCT_MOCKING_LIST };
