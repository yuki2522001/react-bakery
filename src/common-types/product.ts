import { CategoryProps } from "./category";

export interface Product {
  id: string;
  name: string;
  price: number;
  images: string[];
  quantity: number;
  categoryId: string;
  description: string;
  product?: {};
}

export interface ProductContext {
  products: Product[];
  dispatch: Function,
  searchValue: string;
  setSearchValue: Function;
  categories: CategoryProps[];
  setCategories: Function;
}