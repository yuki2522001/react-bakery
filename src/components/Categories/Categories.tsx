import React, { memo } from "react";
import CategoryList from "./CategoryList/CategoryList";

const Categories: React.FC = memo(() => {
  return (
    <div data-testid="categories" className="categories">
      <p className="categories__title">What are you looking for here?</p>
      <CategoryList />
    </div>
  );
});

export default Categories;
