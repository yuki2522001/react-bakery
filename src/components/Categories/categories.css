.categories {
  margin-top: 50px;
  text-align: center;
}

.categories__title {
  text-transform: uppercase;
  font-weight: bold;
}

.categories__list {
  display: flex;
  justify-content: center;
  margin: 30px 0 40px 0;
  padding: 0;
}

.categories__item {
  list-style: none;
  cursor: pointer;
  margin: 0 30px;
  position: relative;
  color: var(--text-secondary);
}

.categories__item:hover {
  color: var(--text-primary);
}

.active {
  color: var(--text-primary);
}

.categories__item > a {
  color: var(--text-third);
  text-decoration: none;
}