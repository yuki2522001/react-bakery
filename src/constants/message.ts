const SUCCESS_MSG = {
  MESSAGE_ADD_PRODUCT: "Product add successful",
  MESSAGE_UPDATE_PRODUCT: "Product update successful",
  MESSAGE_DELETE_PRODUCT: "Product delete successful",
};

const ERROR_MSG = {
  REQUIRED: "Field is required",
  NUMBER: "Enter a non-negative number"
}

export { SUCCESS_MSG, ERROR_MSG };
