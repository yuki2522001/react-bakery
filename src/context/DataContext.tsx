import { ProductContext } from "@common-types/product";
import { createContext, useMemo, useReducer, useState } from "react";
import { productReducer } from "@reducer/productReducer";
import { DataState } from "@common-types/data";
import { CategoryProps } from "@common-types/category";

const initialState: DataState = {
  products: [],
};

const DataContext = createContext<ProductContext>({} as ProductContext);
const DataProvider: React.FC<{children: JSX.Element[] | JSX.Element}> = ({ children }) => {
  const [searchValue, setSearchValue] = useState<string>("");
  const [categories, setCategories] = useState<CategoryProps[]>([]);

  const [state, dispatch] = useReducer(productReducer, initialState);
  const { products } = state;

  const value = useMemo(() => ({
    products,
    dispatch,
    searchValue,
    setSearchValue,
    categories,
    setCategories,
  }), [products, searchValue]);

  return (
    <DataContext.Provider value={value}>
      {children}
    </DataContext.Provider>
  )
}

export {DataProvider, DataContext};