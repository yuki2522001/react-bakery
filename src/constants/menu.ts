import { MenuType } from "../common-types/menu";

const MENU_LIST: MenuType[] = [
  {
    key: "about",
    label: "About",
  },
  {
    key: "shop",
    label: "Shop",
  },
  {
    key: "blog",
    label: "Blog",
  },
  {
    key: "faq",
    label: "FAQ",
  },
  {
    key: "contact",
    label: "Contact"
  }
];

export default MENU_LIST;
