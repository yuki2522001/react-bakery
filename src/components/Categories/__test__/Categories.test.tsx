import "@testing-library/jest-dom";
import Categories from "../Categories";
import mockAxios from "@__mocks__/axios";
import { render } from "@testing-library/react";
import { CATEGORY_MOCKING_LIST } from "@__mocks__/constants/categories";
import { ProductContext } from "@common-types/product";
import { PRODUCT_MOCKING_LIST } from "@__mocks__/constants/product";
import { DataContext } from "@context/DataContext";

const contextProductMock: ProductContext = {
  products: PRODUCT_MOCKING_LIST,
  dispatch: jest.fn(),
  searchValue: "",
  setSearchValue: jest.fn(),
  categories: CATEGORY_MOCKING_LIST,
  setCategories: jest.fn(),
};

describe("Category component", () => {
  test("render category component", async () => {
    mockAxios.get.mockResolvedValueOnce({ data: CATEGORY_MOCKING_LIST });
    const { getByTestId } = render(
      <DataContext.Provider value={contextProductMock}>
        <Categories />
      </DataContext.Provider>
    );
    const categories = getByTestId("categories");
    expect(categories).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <DataContext.Provider value={contextProductMock}>
        <Categories />
      </DataContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
