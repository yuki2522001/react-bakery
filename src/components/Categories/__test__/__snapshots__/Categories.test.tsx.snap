// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="categories"
    data-testid="categories"
  >
    <p
      class="categories__title"
    >
      What are you looking for here?
    </p>
    <div
      class="categories"
      data-testid="category-list"
    >
      <ul
        class="categories__list"
      >
        <li
          class="categories__item active"
          data-testid="category-item"
        >
          All
        </li>
        <li
          class="categories__item inActive"
          data-index="1"
        >
          Cake
        </li>
        <li
          class="categories__item inActive"
          data-index="2"
        >
          Breads
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
